packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -1,80 +1,65 @@
 import { logError, logInfo } from "@/services/logging";
-import { SignerType } from "@/types";
+import { GenerateSignatureDataInput } from "@/types";
 import { SCW_SERVICE, errorToJSON } from "@/utils";
-import { verifyMessage } from "@ambire/signature-validator";
 import {
-    ERC165SessionKeyProvider,
-    KernelSmartContractAccount,
-    KillSwitchProvider,
-    ValidatorMode,
-    ZeroDevEthersProvider,
-    constants,
-    convertEthersSignerToAccountSigner,
+    KernelSmartAccount,
+    createKernelAccount,
+    createKernelAccountClient,
+    createZeroDevPaymasterClient,
 } from "@zerodev/sdk";
-import { ethers } from "ethers";
-import { WalletClient, createWalletClient, custom, getFunctionSelector, publicActions } from "viem";
+import type { KernelAccountClient, KernelValidator } from "@zerodev/sdk";
+import { oneAddress, serializeSessionKeyAccount, signerToSessionKeyValidator } from "@zerodev/session-key";
+import { UserOperation, walletClientToSmartAccountSigner } from "permissionless";
+import { Hex, createWalletClient, custom, http } from "viem";
+import type { EIP1193Provider, PublicClient } from "viem";
+import { Web3 } from "web3";
 
 import { EVMBlockchainIncludingTestnet, chainIdToBlockchain } from "@crossmint/common-sdk-base";
 
 import { CrossmintWalletService } from "../../api/CrossmintWalletService";
-import { GenerateSignatureDataInput } from "../../types/API";
-import { getUrlProviderByBlockchain, getViemNetwork, getZeroDevProjectIdByBlockchain } from "../BlockchainNetworks";
+import { getBundlerRPC, getPaymasterRPC, getUrlProviderByBlockchain, getViemNetwork } from "../BlockchainNetworks";
 import { Custodian } from "../plugins";
-import { TokenType } from "../token/Tokens";
-import BaseWallet from "./BaseWallet";
-import { ZeroDevEip1193Bridge } from "./ZeroDevEip1193Bridge";
-
-type SignerMap = {
-    ethers: ethers.Signer;
-    viem: WalletClient;
-};
-
-export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchainIncludingTestnet> extends BaseWallet {
-    private sessionKeySignerAddress?: string;
+import { TokenType } from "../token";
+
+export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchainIncludingTestnet> {
+    private sessionKeySignerAddress?: Hex;
+    private crossmintService: CrossmintWalletService;
+    private publicClient: PublicClient;
+    private ecdsaValidator: KernelValidator<"ECDSAValidator">;
+    kernelClient: KernelAccountClient;
+    account: KernelSmartAccount;
     chain: B;
 
-    constructor(provider: ZeroDevEthersProvider<"ECDSA">, crossmintService: CrossmintWalletService, chain: B) {
-        super(provider, crossmintService);
+    constructor(
```

Comment:
let's mark this as `private` – right now this is considered a breaking change but clients shouldn't be calling this directly, so let's make sure of that